Model/Push/PushTransactionType.php

```diff
@@ -0,0 +1,465 @@
+            $this->transactionType = $this->pushRequest->getTransactionType();
+            $this->groupTransaction = $this->transactionType === self::BUCK_PUSH_GROUPTRANSACTION_TYPE;
+            $this->creditManagement = $this->pushType === self::BUCK_PUSH_TYPE_INVOICE;
+            $this->magentoServiceAction = $this->pushRequest->getAdditionalInformation('service_action_from_magento');
+            $this->serviceAction = $this->getServiceAction();
+            $this->isFromPayPerEmail = (bool)$this->pushRequest->getAdditionalInformation('frompayperemail')
+                ?? (bool)$this->pushRequest->getAdditionalInformation('frompayperemail')
+                ?? false;
+
+            $this->isSet = true;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Determine the transaction type based on push request data and the saved invoice key.
+     *
+     * @param PushRequestInterface $pushRequest
+     * @param Order $order
+     * @return string
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    public function getPushTypeByInvoiceKey(): string
+    {
+        //If an order has an invoice key, then it should only be processed by invoice pushes
+        $savedInvoiceKey = (string)$this->order->getPayment()->getAdditionalInformation('buckaroo_cm3_invoice_key');
+
+        if (!empty($this->pushRequest->getInvoicekey())
+            && !empty($this->pushRequest->getSchemekey())
+            && strlen($savedInvoiceKey) > 0
+        ) {
+            return self::BUCK_PUSH_TYPE_INVOICE;
+        }
+
+        if (!empty($this->pushRequest->getInvoicekey())
+            && !empty($this->pushRequest->getSchemekey())
+            && strlen($savedInvoiceKey) == 0
+        ) {
+            return self::BUCK_PUSH_TYPE_INVOICE_INCOMPLETE;
+        }
+
+        if (!empty($this->pushRequest->getDatarequest())) {
+            return self::BUCK_PUSH_TYPE_DATAREQUEST;
+        }
+
+        if (empty($this->pushRequest->getInvoicekey())
+            && empty($this->pushRequest->getServiceCreditmanagement3Invoicekey())
+            && empty($this->pushRequest->getDatarequest())
+            && strlen($savedInvoiceKey) <= 0
+        ) {
+            return self::BUCK_PUSH_TYPE_TRANSACTION;
+        }
+
+        return '';
+    }
+
+    /**
+     * Retrieve the status code from the push request based on the transaction type.
+     *
+     * @param string $transactionType
+     * @return int
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    private function getStatusCodeByTransactionType(string $transactionType): int
+    {
+        $statusCode = 0;
+        switch ($transactionType) {
+            case self::BUCK_PUSH_TYPE_TRANSACTION:
+            case self::BUCK_PUSH_TYPE_DATAREQUEST:
+                if ($this->pushRequest->getStatusCode() !== null) {
+                    $statusCode = $this->pushRequest->getStatusCode();
+                }
+                break;
```

Comment:
will make it cleaner if we move the ifs into different function

Reply:
I change it with a shorter version $statusCode = $this->pushRequest->getStatusCode() ?: $statusCode;